lib/shopify_api/rest/base.rb

```diff
@@ -365,10 +365,18 @@ def attributes_to_update
           self.class.read_only_attributes&.include?("@#{attribute}".to_sym)
         end
 
-        HashDiff::Comparison.new(
+        diff = HashDiff::Comparison.new(
           deep_stringify_keys(original_state_for_update),
           deep_stringify_keys(to_hash(true)),
         ).left_diff
+
+        diff.each do |attribute, value|
+          if value.is_a?(Hash) && value[0] == HashDiff::NO_VALUE
+            diff[attribute] = send(attribute)
```

Comment:
What does `send(attribute)` do in this case? I'm not sure why it returns `[]` in this case?

Reply:
`send[attribute]` is how to invoke an instance method from the context of an instance. In this case, we are saying call your method that is named `attribute` in this loop. 

In the context of the test it is kinda like saying `diff["line_items"] = draft_order.line_items` but instead of having a variable named `draft_order` you have the current instance context.

Reply:
https://apidock.com/ruby/Object/send might do a better job than me 😅

Reply:
THANK YOU